lib/Provider/MailService.php

```diff
@@ -0,0 +1,200 @@
+			return (bool) $this->serviceAbilities[$value];
+		}
+		
+		return false;
+
+	}
+
+	/**
+	 * retrieves a collection of what actions a service can perfrom
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return array						collection of abilities otherwise empty collection
+	 */
+	public function capabilities(): array {
+
+		return $this->serviceAbilities;
+
+	}
+
+	/**
+	 * gets the localized human frendly name of this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string						label/name of service (e.g. ACME Company Mail Service)
+	 */
+	public function getLabel(): string {
+
+		return $this->serviceLabel;
+
+	}
+
+	/**
+	 * sets the localized human frendly name of this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $value					label/name of service (e.g. ACME Company Mail Service)
+	 *
+	 * @return self                         return this object for command chaining
+	 */
+	public function setLabel(string $value): self {
+
+		$this->serviceLabel = $value;
+		return $this;
+
+	}
+
+	/**
+	 * gets the primary mailing address for this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return IAddress						mail address object
+	 */
+	public function getPrimaryAddress(): IAddress {
+
+		// retrieve and return primary service address
+		return $this->servicePrimaryAddress;
+
+	}
+
+	/**
+	 * sets the primary mailing address for this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param IAddress $value				mail address object
+	 *
+	 * @return self                         return this object for command chaining
+	 */
+	public function setPrimaryAddress(IAddress $value): self {
+
+		$this->servicePrimaryAddress = $value;
+		return $this;
+
+	}
+
+	/**
+	 * gets the secondary mailing addresses (aliases) collection for this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return array<int, IAddress>			collection of mail address object [IAddress, IAddress]
+	 */
+	public function getSecondaryAddresses(): array {
+
+		// retrieve and return secondary service addressess (aliases) collection
+		return $this->serviceSecondaryAddresses;
+
+	}
+
+	/**
+	 * sets the secondary mailing addresses (aliases) for this service
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param IAddress ...$value				collection of one or more mail address object
+	 *
+	 * @return self                         	return this object for command chaining
+	 */
+	public function setSecondaryAddresses(IAddress ...$value): self {
+
+		$this->serviceSecondaryAddresses = $value;
+		return $this;
+
+	}
+
+	/**
+	 * construct a new fresh message object
+	 *
+	 * @since 30.0.0
+	 *
+	 * @return IMessage				fresh message object
+	 */
+	public function initiateMessage(): IMessage {
+
+		return new Message();
+
+	}
+
+	/**
+	 * sends an outbound message
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param IMessage $message			mail message object with all required parameters to send a message
+	 * @param array $options			array of options reserved for future use
+	 */
+	public function sendMessage(IMessage $message, array $option = []): void {
```

Comment:
```suggestion
	/**
	 * Send an outbound message
	 * 
	 * @since 4.0.0
	 *
	 * @param IMessage $message mail message object with all required parameters to send a message
	 * @param array $option array of options reserved for future use
	 *
	 * @throws \OCP\Mail\Provider\Exception\SendException on failure, check message for reason
	 */
```

- Change $options to $option to match with the method signature
- Add missing exception

Please change $options to $option also for https://github.com/nextcloud/server/blob/0cab17bfe7825e49d3dd6608f20f46164c975e31/lib/public/Mail/Provider/IMessageSend.php#L27.